### PR TITLE
Fix occasional crash on creating overlay tool windows

### PR DESCRIPTION
**Pull request by jherico**
There appears to be an issue with ownership and garbage collection when dynamically creating and destroying tabs in a QML `TabView`.

### CODE REVIEW

**Comment by jherico** [comment]
Addresses https://app.asana.com/0/27650181942747/100742960511126 and possibly https://app.asana.com/0/27650181942747/101905459350855

**Comment by jherico** [comment]
and possibly https://app.asana.com/0/27650181942747/102902798099569

**Comment by hifi-gustavo** [comment]
Build console output for commit: 361d28b686ca8b7dc293f9dfdc02ea74d1ed036c available at: https://gist.github.com/a1da078a8fbcbb9d008d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7422.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7422.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 28f25489f888abbdfbe02f20cdb215410b73834d available at: https://gist.github.com/a1da078a8fbcbb9d008d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7422.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7422.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 7 - Basic smoke tests, nothing odd seen.  Have ran it nearly an hour at this point - opened/closed a lot of QML dialogs and no crashes.  Looks safe enough to turn to wilds for bugsplat input. :+1:

File: interface/resources/qml/ToolWindow.qml [hunk 1]
```diff
@@ -101,21 +94,32 @@ Windows.Window {
         }
     }
 
+    function findFreeTab() {
+        for (var i = 0; i < tabView.count; ++i) {
+            var tab = tabView.getTab(i);
+            if (tab && !tab.originalUrl || tab.originalUrl === "") {
```
Comment (AndrewMeadows) [inline interface/resources/qml/ToolWindow.qml]: According to what I could find about the operator precedence in JavaScript, for example this page:

    https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence

the && operator is evaluated befor the ||.   Which means these two expressions evaluate the same:

`(tab && !tab.originalUrl || tab.originalUrl === "")`

`((tab && !tab.originalUrl) || tab.orignalUrl === "")`

That doesn't look like what you want.

Comment (jherico) [inline interface/resources/qml/ToolWindow.qml]: fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: a30df07d53443e89aaf37680eb5dc6535c3b69c0 available at: https://gist.github.com/a1da078a8fbcbb9d008d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7422.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7422.dmg

**Pull request merged**